scripts/zones/Riverne-Site_A01/Zone.lua

```diff
@@ -18,11 +18,11 @@ end;
 
 function onZoneIn(player,prevZone)
     local cs = -1;
-
-    if (player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0) then
+    if (player:getCurrentMission(COP) == dsp.mission.id.cop.ANCIENT_VOWS and player:getCharVar("PromathiaStatus") == 1) then
```

Comment:
outer parenthesis is not needed. a lot of older scripts still have it and we haven't cleaned all of them yet, sorry for the numerous bad examples.

Also instead of making this the 1st check followed by an else, we probably need this to just be its own check after the original ends. That position 0 check is a safety for landing in the zone without a legit position. If you met the requirements for this conditional but somehow still landed with invalid pos data, you'd get stuck. (I'm aware of your change in the other file setting the pos - but a GM could teleport themselves in without triggering that, as just one example)